libraries/standard/http/cbmc/proofs/Makefile.common

```diff
@@ -0,0 +1,375 @@
+	$(call DO_NOOP_COPY,"Not removing function bodies",$<,$@)
+else
+	$(call DO_GOTO_INSTRUMENT,$(CBMC_REMOVE_FUNCTION_BODY),$<,$@)
+endif
+
+# Don't remove function bodies from proof sources
+$(PROOF_GOTO)1.goto: $(PROOF_GOTO)0.goto
+	$(call DO_NOOP_COPY,"Not removing function bodies",$<,$@)
+
+# Link project and proof sources into the proof harness
+$(ENTRY_GOTO)0.goto: $(PROOF_GOTO)1.goto $(PROJECT_GOTO)1.goto
+	$(call DO_GOTO_CC,--function $(ENTRY),$^,$@)
+
+# Optionally fill static variable with unconstrained values
+$(ENTRY_GOTO)1.goto: $(ENTRY_GOTO)0.goto
+ifeq ($(NONDET_STATIC),"")
+	$(call DO_NOOP_COPY,"Not applying --nondet-static",$<,$@)
+else
+	$(call DO_GOTO_INSTRUMENT,$(CBMC_REMOVE_FUNCTION_BODY) $(NONDET_STATIC),$<,$@)
+endif
+
+# Omit unused functions (sharpens coverage calculations)
+$(ENTRY_GOTO)2.goto: $(ENTRY_GOTO)1.goto
+	$(call DO_GOTO_INSTRUMENT,--drop-unused-functions,$<,$@)
+
+# Omit initialization of unused global variables (reduces problem size)
+$(ENTRY_GOTO)3.goto: $(ENTRY_GOTO)2.goto
+	$(call DO_GOTO_INSTRUMENT,--slice-global-inits,$<,$@)
+
+# Final name for proof harness
+$(ENTRY_GOTO).goto: $(ENTRY_GOTO)3.goto
+	cp $< $@
+
+goto: $(ENTRY_GOTO).goto
+	echo $(DEPENDENT_GOTOS)
+
+################################################################
+# Targets to run the analysis commands
+
+logs/property.txt: $(ENTRY_GOTO).goto
+	$(call DO_CBMC,$(CBMCFLAGS) $(CHECKFLAGS) --trace,$<,$@)
+
+logs/property.xml: $(ENTRY_GOTO).goto
+	$(call DO_CBMC,$(CBMCFLAGS) $(CHECKFLAGS) --unwinding-assertions --trace --xml-ui,$<,$@)
+
+logs/property-list.xml: $(ENTRY_GOTO).goto
+	$(call DO_CBMC,$(CBMCFLAGS) $(CHECKFLAGS) --unwinding-assertions --show-properties --xml-ui,$<,$@)
+
+logs/coverage.xml: $(ENTRY_GOTO).goto
+	$(call DO_CBMC,$(CBMCFLAGS) $(COVERFLAGS) --cover location --xml-ui,$<,$@)
+
+property: logs/property.txt
+
+property-list: logs/property-list.xml
+
+coverage: logs/coverage.xml
+
+report: logs/property.txt logs/property-list.xml logs/coverage.xml
+	$(VIEWER) \
+	--block logs/coverage.xml \
+	--goto $(ENTRY_GOTO).goto \
+	--htmldir html \
+	--property logs/property-list.xml \
+	--result logs/property.txt \
+	--srcdir $(SRCDIR)
+
+# Caution: run make-sources before running property and coverage checking
+# The current make-sources script removes the goto binary
+logs/sources.json:
+	$(MAKE_SOURCES) --srcdir $(SRCDIR) --gotodir $(PROOFDIR)
+
+report2: logs/sources.json logs/property.xml logs/property-list.xml logs/coverage.xml
+	$(VIEWER2) \
+	--make-sources logs/sources.json \
+	--result logs/property.xml \
+	--coverage logs/coverage.xml \
+	--property logs/property-list.xml \
+	--srcdir $(SRCDIR) \
+	--goto $(ENTRY_GOTO).goto \
+	--htmldir html \
+
+################################################################
+# Targets to clean up after ourselves
+clean:
+	$(RM) *.goto
+	$(RM) $(DEPENDENT_GOTOS)
+	$(RM) *.log
+	$(RM) TAGS*
+	$(RM) *~ \#*
+
+veryclean: clean
+	$(RM) -r html
+	$(RM) -r logs
+	$(RM) -r gotos
```

Comment:
`$(PROOFDIR)/html`, `$(LOGDIR)`, `$(GOTODIR)`. Speaking of which, you may wish to define `$(REPORTDIR)`.